builder/azure/arm/config.go

```diff
@@ -105,6 +105,13 @@ type SharedImageGalleryDestination struct {
 	SigDestinationStorageAccountType string `mapstructure:"storage_account_type"`
 }
 
+type Spot struct {
+	// Specify eviction policy for spot instance: "Deallocate" or "Delete". If this is set, a spot instance will be used.
+	EvictionPolicy string `mapstructure:"eviction_policy"`
```

Comment:
I might have missed it in this PR but can we input validation this string? I think this will currently allow you to send anything, I think we should throw an error in Packer if the user tries to send anything other than Deallocate or Delete

Reply:
I adjusted the type and added validation. I'm not *that* familiar with the code base - can you please double check if that's fine or if I should use another approach (and maybe point me to an example I can use as a base in this case?). Thanks!